oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/base/StorageEsInstaller.java

```diff
@@ -38,68 +42,165 @@
 @Slf4j
 public class StorageEsInstaller extends ModelInstaller {
     private final Gson gson = new Gson();
-
     private final StorageModuleElasticsearchConfig config;
     protected final ColumnTypeEsMapping columnTypeEsMapping;
 
+    /**
+     * The mappings of the template .
+     */
+    private final Map<String, Map<String, Object>> tables;
```

Comment:
We still need to polish this part. `Map<String, Map<String, Object>>`->`Map<String, IndexFields>`.
`class IndexFields` could include a map to represent the data structure originally from ES.